src/Avalonia.Controls/ThemeVariantIcons.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.IO;
+using Avalonia.Platform;
+
+namespace Avalonia.Controls;
+
+/// <summary>
+/// Represents two icons, one for use with a light theme and the other for use with a dark theme.
+/// </summary>
+/// <remarks>
+/// A window's icon can appear in multiple locations (e.g. in the window title bar and the platform taskbar), and these locations 
+/// can use different themes. This means that both light and dark icons can be visible at the same time.
+/// </remarks>
+public class ThemeVariantIcons : WindowIcon
```

Comment:
`ThemedWindowIcon` might be a better name here. The name as you have it is quite literal which is generally a good thing but doesn't quite fit with other icon control terminology.

In the end only one theme variant is actually shown too: either light or dark. So while the class contains both representations internally it only represents one icon on screen.

Reply:
Adding "Window" to the name sounds good to me. I'm less keen on removing "Variant". The icons relate to `ThemeVariant` (well, actually the `PlatformThemeVariant` enum), not themes themselves like Fluent or Simple.

BTW it's not true that only one icon is ever shown. Check the XML documentation and/or PR description.

Reply:
> I'm less keen on removing "Variant". The icons relate to ThemeVariant (well, actually the PlatformThemeVariant enum)

Fair, I still don't like the "theme variant" terminology but it is what it is. The name is quite verbose in this case and I wonder if we can side-step it completely. "AdaptiveWindowIcon", etc.

> BTW it's not true that only one icon is ever shown

This is just semantics I think. I'm saying only one variant of the icon is shown at a time in any given location. So you are essentially putting a control in that can switch it appearance based on theme variant automatically. We don't make terminology plural for this though. A dark expander is the same as a light expander even though some of the resources it uses to draw itself are different. In my mind this theme variant aware icon is equivalent.

This is also somewhat loosely related to the discussions here: https://github.com/microsoft/microsoft-ui-xaml/issues/2257

Reply:
I don't think there is an English word which means "one of two modes". `ThemeVariantWindowIcon` is the best I can come up with, without the name becoming vague.